cfpb/cfgov-refresh
# For discussion: Changing our approach to browser tests

We'd like to make it easier for front-end developers to write browser tests. In the current setup, one of the biggest barriers to this is creating test data via Python. Secondarily, the time it takes to fire up `tox` to run the tests can be annoying, if your environment is out of date.

As most of our FEWDs are accustomed to running a full local environment for testing already, @contolini and I thought, "Why not just have the browser tests run against an already-running local server?"

This PR begins to implement that notion.

## Testing

1. Pull branch
1. Get a local server up and running
1. Ensure that the following env vars are set:
   - `export TEST_HTTP_HOST=localhost`
   - `export DJANGO_HTTP_PORT=8000`
1. Run `gulp test:acceptance:protractor`

## Notes

- Removing the ability to test with `tox` necessitates taking that step out of Travis, which has the side benefit of saving several minutes per Travis build

## Todos

- [ ] Remove all the tox cruft and simplify the `gulp` command to just `gulp test:acceptance`
- [ ] Simplify some of the common JS & config
- [ ] Add the two env vars mentioned above to our default env var lists in the sample files (reconcile with #4391)
- [ ] Modify Travis scripts to remove this step
- [ ] Set up post-deploy smoke tests against Beta in EXT Jenkins
- [ ] Figure out why the protractor-retry functionality isn't working as expected

## Checklist
